src/main/java/io/github/wimdeblauwe/errorhandlingspringbootstarter/mapper/ErrorCodeMapper.java

```diff
@@ -38,6 +34,24 @@ public String getErrorCode(Throwable exception) {
         }
     }
 
+    private String getErrorCode(Class<?> exceptionClass) {
```

Comment:
This can be `Class<? extends Throwable>`

Reply:
As Throwable is an interface I could have a hierarchy like this:

- public class MyRootClass extends Object;
- public class MiSecondLevelClass extends MyRootClass implements Throwable;

Here I have two options.  Allow look throw the full class hierarchy (which is what I'm doing right now) or only keep looking while the parent class is also a Throwable (which I should check before going up in the hierarchy).

If I use Class<? extends Throwable> then I need to check if parent class is assignable from Throwable and then cast the recursive call like these (supressing the unchecked warning):
```java
        Class<?> superClass = exceptionClass.getSuperclass();
        if (Throwable.class.isAssignableFrom(superClass)) {
            @SuppressWarnings("unchecked") Class<? extends Throwable> superThrowable = (Class<? extends Throwable>) superClass;
            return getErrorCode(superThrowable);
        } else {
            return null;
        }
```
@wimdeblauwe  Is this what you want?

Reply:
Ah, you are right, there is no need that you need to use a class that extends from Throwable. A bit strange, but possible. Let's keep as is indeed. 👍